Tests/PhishingDetectionTests/PhishingDetectionUpdateManagerTests.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  PhishingDetectionUpdateManagerTests.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+import XCTest
+@testable import PhishingDetection
+
+class PhishingDetectionUpdateManagerTests: XCTestCase {
+    var updateManager: PhishingDetectionUpdateManager!
+    var dataStore: PhishingDetectionDataSaving!
+    var mockClient: MockPhishingDetectionClient!
+
+    override func setUp() {
+        super.setUp()
+        mockClient = MockPhishingDetectionClient()
+        dataStore = MockPhishingDetectionDataStore()
+        updateManager = PhishingDetectionUpdateManager(client: mockClient, dataStore: dataStore)
+    }
+
+    override func tearDown() {
+        updateManager = nil
+        dataStore = nil
+        mockClient = nil
+        super.tearDown()
+    }
+
+    func testUpdateHashPrefixes() async {
+        await updateManager.updateHashPrefixes()
+        XCTAssertFalse(dataStore.hashPrefixes.isEmpty, "Hash prefixes should not be empty after update.")
+        XCTAssertEqual(dataStore.hashPrefixes, [
+            "aa00bb11",
+            "bb00cc11",
+            "cc00dd11",
+            "dd00ee11",
+            "a379a6f6"
+        ])
+    }
+
+    func testUpdateFilterSet() async {
+        await updateManager.updateFilterSet()
+        XCTAssertFalse(dataStore.filterSet.isEmpty, "Filter set should not be empty after update.")
+    }
+
+    func testRevision1AddsData() async {
```

Comment:
I think for these test we should test the final result is are the expected list (original plus added minus removed)
Also to do this you actually have always to do the initial update.
Therefore I would add it to the setup